lib/gis_robot_suite/descriptive_metadata_builder.rb

```diff
@@ -0,0 +1,595 @@
+        'gmd:aggregationInfo/gmd:MD_AggregateInformation/gmd:aggregateDataSetName/gmd:CI_Citation/gmd:otherCitationDetails/gco:CharacterString[contains(text(), "Related publication")]', NS
+      )
+      return unless related_publications.any?
+
+      related_publications.map do |node|
+        { type: 'citation/reference',
+          displayLabel: 'Related publication',
+          value: related_citation(node) }
+      end
+    end
+    # rubocop:enable Layout/LineLength
+
+    def related_citation(node)
+      authors = node.xpath('../../gmd:citedResponsibleParty/gmd:CI_ResponsibleParty/gmd:individualName/gco:CharacterString', NS).collect(&:text)
+                    .concat(node.xpath('../../gmd:citedResponsibleParty/gmd:CI_ResponsibleParty/gmd:organisationName/gco:CharacterString', NS).collect(&:text))
+      authors = authors.join(', ').chomp('.').concat('.') # Remove any existing ending middle initial period before adding period to end of authors string
+      title = "\"#{node.xpath('../../gmd:title/gco:CharacterString', NS).text}.\"" # "Example article title"
+      pub_date = node.xpath('../../gmd:date/gmd:CI_Date/gmd:date/gco:Date', NS).text
+      pub_year = pub_date.present? ? "(#{Date.parse(pub_date).strftime('%Y')})." : nil # Year in parens (2023)
+      series = node.xpath('../../gmd:series/gmd:CI_Series/gmd:name/gco:CharacterString', NS).text
+      identifier = node.xpath('../../gmd:identifier/gmd:MD_Identifier/gmd:code/gco:CharacterString', NS)
+      identifier_text = identifier.any? ? "Available at: #{identifier.text}" : nil
+
+      [authors, title, series, pub_year, identifier_text].compact.join(' ')
+    end
+
+    def admin_metadata
+      {}.merge({ contributor: [{ name: [{ value: 'Stanford' }] }] })
+        .merge(admin_identifier)
+        .merge(admin_langs)
+        .compact
+    end
+
+    def admin_identifier
+      identifier = iso19139_ng.xpath('//gmd:MD_Metadata/gmd:fileIdentifier/gco:CharacterString', NS)
+      raise "identifier not found in '//gmd:MD_Metadata/gmd:fileIdentifier/gco:CharacterString'" unless identifier.any?
+
+      { identifier: [{ value: identifier.text }] }
+    end
+
+    def admin_langs
+      langs = iso19139_ng.xpath('//gmd:MD_Metadata/gmd:language/gmd:LanguageCode', NS)
+      return unless langs.any?
+
+      data = langs.map do |lang|
+        { code: lang.xpath('@codeListValue', NS).text,
+          source: { code: lang.xpath('@codeSpace', NS).text } }
+      end
+
+      { language: data }
+    end
+
+    def access
+      access_nodes = data_id_node.xpath(
+        'gmd:pointOfContact/gmd:CI_ResponsibleParty/gmd:contactInfo/gmd:CI_Contact/gmd:address/gmd:CI_Address/gmd:electronicMailAddress/gco:CharacterString', NS
+      )
+      return unless access_nodes.any?
+
+      { accessContact: access_nodes.map do |node|
+                         { value: node.text, type: 'email', displayLabel: 'Contact' }
+                       end }
+    end
+
+    def rootdir
```

Comment:
Code from here to end is not new and is brought over from generate-mods. I made some minor updates to variable names (e.g. `w` -> `west`)